### DISCUSSION TITLE
Roadmap / todo list

### DISCUSSION THREAD

**Discussion by jhelvy** [surveydown-dev/surveydown#156] [Announcements]
We posted a "To Do" list repo for the project [here](https://github.com/surveydown-dev/todo). As we build out new ideas, we'll update the [todo.csv file](https://github.com/surveydown-dev/todo/blob/main/todo.csv) in that repo. 

For any new feature requests, please DO NOT comment here but instead [start a new discussion](https://github.com/orgs/surveydown-dev/discussions/new/choose) and we'll tag it in the todo repo.

**Comment by pingfan-hu** [comment]
# TODO List

- Wide variety of [question types](https://surveydown.org/question-types) referring to the [official widgets from shiny](https://shiny.posit.co/r/gallery/widgets/widget-gallery/), and the [shinyWidgets package by dreamRs](https://github.com/dreamRs/shinyWidgets).
  - [x] `text` - Text input
  - [x] `textarea` - Text area input
  - [x] `numeric` - Numeric input
  - [x] `mc` - Multiple choice type with single choice
  - [x] `mc_multiple` - Multiple choice type with multiple selections
  - [x] `mc_buttons` - Button version of `mc`
  - [x] `mc_multiple_buttons` - Button version of `mc_multiple `
  - [x] `select` - Dropdown select type
  - [x] `slider ` - Slider type, supporting both numeric and text labels
  - [x] `date` - Date input type
  - [x] `daterange` - Date range input type
  - [x] `matrix` - Matrix input type that embraces multiple `mc` questions
  - [ ] Best worst
  - [ ] Single checkbox with on and off toggle switches
  - [ ] Select menu - Dropdown select type with multiple selections
  - [ ] Slider with no labels
  - [ ] Tree inputs
- [x] Custom question type - A question placeholder for customized questions
- [x] Ability to [use latest survey results](https://surveydown.org/fetch-data#reactive-fetching) in the survey itself
- [x] Ability to [Ignore the database connection](https://surveydown.org/store-data#ignoring-the-supabase-connection)
- [x] Ability to [conditionally display questions](https://surveydown.org/conditional-control#conditional-display)
- [x] Ability to [conditionally skip to pages](https://surveydown.org/conditional-control#conditional-skipping)
- [x] Start the survey from a [specific page](https://surveydown.org/server-options#starting-page) (helpful when editing survey)
- [x] [Require](https://surveydown.org/server-options#required-questions) specific questions or all questions be answered
- [x] Create a random numeric [completion code](https://surveydown.org/reactivity#displaying-stored-values-e.g.-a-completion-code)
- [x] Time stamps recorded for each question and page interaction
- [x] [Auto scroll](https://surveydown.org/server-options#auto-scroll) according to the answering progress
- [x] [Progress bar](https://surveydown.org/survey-components#progress-bar) bar that updates on each question interaction
- [x] Customizable [progress bar](https://surveydown.org/survey-components#progress-bar) color and position on page
- [x] [Markdown formatting](https://surveydown.org/question-formatting#markdown-formatting) for options and buttons
- [x] [Support for bootstrap themes](https://surveydown.org/survey-components#changing-the-look-and-feel)
- [x] [Customizable `scss` theme file](https://surveydown.org/survey-components#changing-the-look-and-feel)
- [x] [Pass parameters](https://surveydown.org/redirect#reactive-redirect) through the url, e.g. to track user IDs
- [x] [Redirect](https://surveydown.org/redirect) users to an external url
- [x] Store session ID in [browser cookies](https://surveydown.org/server-options#cookies) to store user progress if they refresh the page
- [ ] [Admin page](https://surveydown.org/server-options#admin-page) with password login to preview / download data, pause survey, etc.
- [ ] Form validation (limit input based on question type)
